### PR TITLE
contrib: use user.ghtoken for github api requests

### PR DESCRIPTION
**Pull request by glozow**
Avoid 403 rate limit errors

### CODE REVIEW

File: contrib/new_post.py [hunk 1]
```diff
@@ -54,6 +54,15 @@ class PullRequest():
     labels: list[str]
     user: str
 
+def git_config_get(option, default=None):
+    '''
+    Get named configuration option from git repository.
+    '''
+    try:
+        return subprocess.check_output(['git','config','--get',option]).rstrip().decode('utf-8')
```
Comment (jnewbery) [inline contrib/new_post.py:62]: `check_output()` is the old high-level API: https://docs.python.org/3/library/subprocess.html#older-high-level-api , and most new code uses `subprocess.run()`. Any particular reason to go for `check_output()` here?

**Review by jnewbery — APPROVED** [review]
LGTM! ACK 88501ff34b06fc54814a7523787e83c08c21a5bf

File: contrib/new_post.py (hunk 1 above)
Comment (glozow) [reply contrib/new_post.py]: I ripped it from another script. Agree `run` seems preferred, changed.

**Comment by stickies-v** [comment]
I think this is buggy when `user.ghtoken` is not set:

```
% ./contrib/new_post.py 30239 tdb3 2024-11-06
Github returned error HTTP Error 401: Unauthorized
```

`subprocess.run()` does not throw `CalledProcessError` for non-zero exit codes, that's `subprocess.check_output()` behaviour. As a result, without `user.ghtoken` set, an empty-string `Authorization` token is being added, leading to a `401`.

Will open a pull to fix.

**Pull request merged**